### PR TITLE
get codemirror working in this environment

### PR DESCRIPTION
**Pull request by jlongster**
A few fixes I had to do to get it working for the debugger (should fix for other tools also)

### CODE REVIEW

File: client/sourceeditor/codemirror.html [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<!DOCTYPE html>
+<html dir='ltr'>
+  <head>
+    <style>
+      html, body { height: 100%; }
+      body { margin: 0; overflow: hidden; }
+      .CodeMirror { width: 100%; height: 100% !important; line-height: 1.25 !important;}
+    </style>
+    <link rel="stylesheet" type="text/css" href="/client/themes/common.css" />
```
Comment (bgrins) [inline client/sourceeditor/codemirror.html]: Good idea, using a separate page makes resolving these paths much easier.  Can you please make all these paths relative to the html page instead? so `/client/sourceeditor/codemirror/` -> `codemirror/` and `/client/shared/theme-switching.js` -> `../shared/theme-switching.js`.

File: client/sourceeditor/editor.js [hunk 2]
```diff
@@ -46,10 +46,6 @@ const { OS } = Services.appinfo;
 // order to initialize a CodeMirror instance.
 
 const CM_STYLES   = [
-  "/devtools/client/themes/common.css",
-  "/devtools/client/sourceeditor/codemirror/lib/codemirror.css",
-  "/devtools/client/sourceeditor/codemirror/addon/dialog/dialog.css",
-  "/devtools/client/sourceeditor/codemirror/mozilla.css"
 ];
 
 const CM_SCRIPTS  = [
```
Comment (bgrins) [inline client/sourceeditor/editor.js]: Could all of these CM_SCRIPTS move to the HTML page instead (as script tags), and all the baseURI / scriptLoader stuff in this file can be removed?  I don't care so much about refactoring it as much as I don't think concatenating baseURI with each of these scripts will come up with the right path, and it's easier to come up with the right path from the HTML page

File: client/sourceeditor/codemirror.html (hunk 1 above)
Comment (jlongster) [inline client/sourceeditor/codemirror.html]: ok

File: client/sourceeditor/editor.js (hunk 2 above)
Comment (jlongster) [inline client/sourceeditor/editor.js]: Yeah, that sounds good.

**Comment by jlongster** [comment]
Ok, here is the new code plus a few more fixes I had to do because of broken changes that were committed since last I worked on it.

**Pull request merged**